test/run_tests.pl

```diff
@@ -39,6 +39,11 @@ BEGIN
 $ENV{OPENSSL_ENGINES} = rel2abs(catdir($bldtop, "engines"));
 $ENV{CTLOG_FILE} = rel2abs(catfile($srctop, "test", "ct", "log_list.cnf"));
 
+#On platforms that support this, this will ensure malloc returns data that is
+#set to a non-zero value. Can be helpful for detecting uninitialized reads in
+#some situations.
+$ENV{'MALLOC_PERTURB_'} = '128';
```

Comment:
How does this interact with asan?  Memory will always be initialised which cuts out a number of detections.

Reply:
Note that asan doesn't detect uninit reads anyway. That's what msan does. However I ran some tests to check the interaction between this and asan, msan and valgrind.

In all cases I found that this environment variable is ignored if you use any of those 3 tools. The output from malloc is not perturbed. Only if you don't use any of them does it seem to have an effect.

Reply:
I guess you should not override any MALLOC_PERTURB from the user here.

Reply:
Done